ark/utils/deepcell_service_utils.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+from twisted.internet import reactor
+from kiosk_client import manager
+import os
+from pathlib import Path
+import shutil
+
+
+def create_deepcell_output(deepcell_input_dir, points, deepcell_output_dir,
+                           host='https://deepcell.org', job_type='multiplex'):
+    """ Handles all of the necessary data manipulation for running deepcell tasks.
+
+        Creates .zip files (to be used as input for DeepCell),
+        calls run_deepcell_task method,
+        and extracts zipped output files to the specified output location
+
+        Args:
+            points (list):              List of points in preprocessing pipeline
+
+            deepcell_input_dir (str):   Location of preprocessed files
+                                        (assume deepcell_input_dir contains <point>.tif
+                                        for each point in points list)
+
+            deepcell_output_dir (str):  Location to save DeepCell output (as .tif)
+
+            host:                       Hostname and port for the kiosk-frontend API server
+                                        Default: 'https://deepcell.org'
+
+            job_type:                   Name of job workflow (multiplex, segmentation, tracking)
+                                        Default: 'multiplex'
+
+        Output:
+            Writes DeepCell service .tif output to output_dir
+
+        Raises:
+            UserWarning:                Raised if DeepCell output .zip file
+                                        was not created for some point in points list.
+
+        """
+
+    # create /zip folders if they do not exist, and make sure they are empty
+    input_zip = os.path.join(deepcell_input_dir, 'zip')
+    output_zip = os.path.join(deepcell_output_dir, 'zip')
+    for directory in [input_zip, output_zip]:
+        Path(directory).mkdir(parents=True, exist_ok=True)
+        [f.unlink() for f in Path(directory).glob("*") if f.is_file()]
+
+    # zip the preprocessed .tif files
+    for i, point in enumerate(points):
```

Comment:
If you give the kiosk-client a single zip file with multiple TIFs in it, it will analyze all of the files inside the zip.
I think this will help simplify the code. We can just create a single zip file in the same folder that has .tiff files, and submit that single zip file to the client. Then we can unzip the output directly into the output directory

Reply:
Definitely makes it easier, thanks.
updated In the last commit.